packages/transfer-app/src/Saved.tsx

```diff
@@ -0,0 +1,155 @@
+};
+
+export class Saved extends React.PureComponent<Props, State> {
+  static contextType = ApiContext;
+
+  context!: React.ContextType<typeof ApiContext>; // http://bit.ly/typescript-and-react-context
+
+  state: State = {
+    allAccounts: [],
+    allAddresses: []
+  };
+
+  memo: Memo = {};
+
+  componentDidMount () {
+    const { keyring } = this.context;
+
+    this.setState({
+      allAccounts: keyring.getPairs(),
+      allAddresses: keyring.getAddresses()
+    });
+  }
+
+  render () {
+    return (
+      <WalletCard
+        header='Saved Identities'
+        overflow='scroll'
+        subheader='Quickly select an identity that you have previously saved to transfer balance to'>
+        <MarginTop marginTop='3.5rem' />
+        <Grid>
+          <Grid.Column width={8}>
+            <Stacked>
+              <SubHeader> My Unlocked Accounts </SubHeader>
+              <WithSpace>
+                  { this.renderAccountsToSendFrom() }
+              </WithSpace>
+            </Stacked>
+          </Grid.Column>
+          <Grid.Column width={8}>
+            <Stacked>
+              <SubHeader> Saved Addresses </SubHeader>
+              <WithSpace>
+                  { this.renderAddressesToSendTo() }
+              </WithSpace>
+            </Stacked>
+          </Grid.Column>
+        </Grid>
+      </WalletCard>
+    );
+  }
+
+  renderAccountsToSendFrom () {
+    const { allAccounts } = this.state;
+
+    if (!allAccounts.length) {
+      this.renderEmpty();
+    }
+
+    return allAccounts.map((pair: KeyringPair) => {
+      return (
+        <React.Fragment key={`__unlocked_${pair.address()}`}>
+          <MarginTop />
+          <Link to={`/transfer/${pair.address()}`}>
+            <AddressSummary
+              address={pair.address()}
+              name={pair.getMeta().name}
+              orientation='horizontal'
+              size='small' />
+          </Link>
+        </React.Fragment>
+      );
+    });
+  }
+
+  // Note: currying to pass value onClick without a closure:
+  // https://www.sitepoint.com/currying-in-functional-javascript/
+  handleSelectedRecipient = (address: string) => {
+    const { onSelectAddress } = this.props;
+
+    const handler = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      e.preventDefault();
+      onSelectAddress(address);
+    };
+
+    if (!this.memo[address]) {
+      this.memo[address] = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => handler(e);
+    }
+
+    return this.memo[address];
+  }
+
+  renderAddressesToSendTo () {
+    const { allAddresses } = this.state;
+
+    if (!allAddresses.length) {
+      this.renderEmpty();
+    }
+
+    return allAddresses.map((address: KeyringAddress) => {
+      return (
+        <React.Fragment key={`__locked_${address.address()}`}>
+          <MarginTop />
+          <StyledLinkButton onClick={this.handleSelectedRecipient(address.address())}>
```

Comment:
This will still create a new closure (the `const handler = ...` closure) on each render, so won't solve the performance issue with `onClick={(e) => {/* do something with e and address */}}`

There are some solutions (create a new component, use `data-*`) but I think it's alright to use here a closure `onClick={(e) => {/* do something with e and address */}}`, and it's a bit clearer to read than a currified function.